File: Requests.cs
using ChatReciver.utils;
using Newtonsoft.Json.Linq;

namespace ChatReciver
{
    internal class Requests
    {
        public static async Task<ChatData> FetchChat(LivePageData options)
        {
            string url = "https://www.youtube.com/youtubei/v1/live_chat/get_live_chat?key=" + options.apiKey;
            JObject body = new JObject();
            JObject context = new JObject();
            JObject client = new JObject();

            client["clientVersion"] = options.clientVersion;
            client["clientName"] = "WEB";

            context["client"] = client;
            body["context"] = context;
            body["continuation"] = options.continuation;

            JObject res = await new HTTPManager(ChatReciver.Program.client).PostAsyncJObject(url, body.ToString());
            //Console.WriteLine(res.ToString());
            return Parser.ParseChatData(res);
            //return null;
        }

        //Done
        public static async Task<LivePageData> FetchLivePage(string id)
        {
            string url = GenerateLiveUrl(id) ?? throw new Exception("ID not avaliable");
            string resp = await new HTTPManager(Program.client).GetAsyncString(url);

            return Parser.GetOptionsFromLivePage(resp, id);
        }

        //Done
        public static string GenerateLiveUrl(string YoutubeId)
        {
            return "https://www.youtube.com/watch?v=" + YoutubeId;
        }
    }
}

File: api/src/main/java/cz/iqlandia/iqplanetarium/chatreciever/Requests.java
package cz.iqlandia.iqplanetarium.chatreciever;

import cz.iqlandia.iqplanetarium.chatreciever.records.ChatData;
import cz.iqlandia.iqplanetarium.chatreciever.utils.HTTPManager;
import org.json.JSONObject;

import java.io.IOException;
import java.text.ParseException;

/**
 * Analog to requests.ts
 */
public class Requests {

    public static LivePageData fetchLivePage(String id) throws IOException, ParseException {
        return Parser.getLivePageData(HTTPManager.getHTTPString("https://www.youtube.com/watch?v="+id));
    }
    public static ChatData fetchChat(LivePageData data) throws IOException {
        //System.out.println(data);
        String url = "https://www.youtube.com/youtubei/v1/live_chat/get_live_chat?key=" + data.apiKey;
        JSONObject body = new JSONObject();
        body.put("context", new JSONObject()
                .put("client", new JSONObject()
                        .put("clientVersion", data.clientVersion)
                        .put("clientName", "WEB")));
        body.put("continuation", data.continuation);

        JSONObject res = HTTPManager.postHTTPJSONObject(url, body);
        //System.out.println(res.toString(4));
        return Parser.parseChatData(res);
    }
}
